preview/MsixCore/msixmgr/AutoPlay.cpp

```diff
@@ -0,0 +1,577 @@
+    // a 21 byte buffer. A 21 byte buffer translates into 34 characters. Although
+    // this meets the requirements, it is confusing since a 33 character limit
+    // results in a 34 character long encoding. In comparison, a 32 character long
+    // encoding divides evenly and always results in a 20 byte digest.
+
+    static const size_t MaxBase32EncodedStringLength = 32;
+
+    // The maximum number of bytes the digest can be is 20. We arrive at this by:
+    // - The maximum count of characters [without prefix] in the encoding (32):
+    // - multiplied by 5 (since each character in base 32 encoding is based off of 5 bits)
+    // - divided by 8 (to find how many bytes are required)
+    // - The  initial plus 7 is to enable integer math (equivalent of writing ceil)
+
+    static const ULONG MaxByteCountOfDigest = (MaxBase32EncodedStringLength * 5 + 7) / 8;
+
+    HRESULT hr = S_OK;
+
+    // Build the progIdSeed by appending the incoming strings
+    // The package moniker and the application ID are case sensitive
+    //std::wstring tempProgIDBuffer;
+
+    std::wstring tempProgIDBuilder;
+    tempProgIDBuilder.append(packageMoniker);
+    std::transform(tempProgIDBuilder.begin(), tempProgIDBuilder.end(), tempProgIDBuilder.begin(), ::tolower);
+    tempProgIDBuilder.append(applicationId);
+
+    // The category name and the subcategory are not case sensitive
+    // so we should lower case them
+
+    std::wstring tempLowerBuffer;
+    tempLowerBuffer.assign(categoryName);
+    std::transform(tempLowerBuffer.begin(), tempLowerBuffer.end(), tempLowerBuffer.begin(), ::tolower);
+    tempProgIDBuilder.append(tempLowerBuffer);
+
+    if (!subCategory.empty())
+    {
+        tempLowerBuffer.assign(subCategory);
+        std::transform(tempLowerBuffer.begin(), tempLowerBuffer.end(), tempLowerBuffer.begin(), ::tolower);
+        tempProgIDBuilder.append(tempLowerBuffer);
+    }
+
+    // Create the crypto provider and start the digest / hash
+    AutoPtr<CryptoProvider> cryptoProvider;
+    RETURN_IF_FAILED(CryptoProvider::Create(&cryptoProvider));
+    RETURN_IF_FAILED(cryptoProvider->StartDigest());
+    COMMON_BYTES data = { 0 };
+    data.length = (ULONG)tempProgIDBuilder.size() * sizeof(WCHAR);
+    data.bytes = (LPBYTE)tempProgIDBuilder.c_str();
+
+    RETURN_IF_FAILED(cryptoProvider->DigestData(&data));
+
+    // Grab the crypto digest
+    COMMON_BYTES digest = { 0 };
+    RETURN_IF_FAILED(cryptoProvider->GetDigest(&digest));
+
+    // Ensure the string buffer has enough capacity
+    StringBuffer base32EncodedDigest;
+    RETURN_IF_FAILED(base32EncodedDigest.SetCapacity(
+        MaxBase32EncodedStringLength));
+
+    // Base 32 encode the bytes of the digest and put them into the string buffer
+    ULONG base32EncodedDigestCharCount = 0;
+    RETURN_IF_FAILED(Base32Encoding::GetChars(
+        digest.bytes,
+        GetMinValue(digest.length, MaxByteCountOfDigest),

```

Comment:
>GetMinValue [](start = 8, length = 11)

does std::min work?

Reply:
Yes, this works, updated to use std::min and remove the GetMinValue method.